### ISSUE TITLE
Present the draft Fraud Risk Management architecture to the Design Authority for review

### ISSUE THREAD

**Issue by JustusOrtlepp** [mojaloop/design-authority-project#73]
## Request Summary: 

We have composed a draft architecture for a Fraud Risk Management solution that facilitates the collection of transaction information destined for the Mojaloop platform and the subsequent evaluation of those transactions for fraud and money laundering risk. We'd like to present the draft architecture to the Design Authority for review, discussion and comment.

## **Request Details**:

- **Deadline:** 31 January 2021 - we intend to commence development of the MVP in February (PI14) after the completion of the PoC (PI13)
- **Impact (Teams):** Assistance is required to identify impacted teams based on the draft architecture.
- **Impact (Components):** Assistance is required to identify impacted components (e.g. APIs, docs, codebases) based on the draft architecture.

The Fraud Risk Management platform intends to screen all transactions submitted to a Mojaloop switching hub in either real-time or near-real-time, depending on risk appetite and operational needs. Transactions must be ingested by the FRM solution at the same pace as those processed by the Mojaloop platform itself. We have explored a number of architectural designs that aim to achieve and maintain the performance goals of both platforms and would like to present our draft architecture for review.

**Artifacts**:
- [ ] Artifact to consider [@Contributor]

**Dependencies**:
- [ ] If Applicable

### **Accountability**:
- **Owner:** <DA member responsible for issue>
- **Raised By:** Justus Ortlepp-

## **Decision(s)**:

<fill in this section upon reaching a decision>

- **Approved By:** <Insert name>
  
### Details

- [ ] Actual decision made as a result of discussion

## **Follow-up**:

<fill in this section upon reaching a decision>
  
- [ ] Actions to implement the decisions

**Comment by godfreykutumela** [comment]
Hi @JustusOrtlepp do you perhaps we have any prereading material on this and also please confirm how much time do you need for this? @bushjames @simeonoriko

**Comment by JustusOrtlepp** [comment]
> Hi @JustusOrtlepp do you perhaps we have any prereading material on this and also please confirm how much time do you need for this? @bushjames @simeonoriko

Hi @godfreykutumela,

There are three options in managing the integration between Mojaloop and the FRMS:

1. The FRMS will act as an intercepting proxy that wraps around a Mojaloop implementation (lowest deployment impact and facilitates a more open architecture for integration with other platforms/switches)
2. As a service that responds to transaction PUTS from Mojaloop within Kafka.
3. As a gateway that replaces the current Mojaloop gateway for a seamless drop-in replacement gateway module
 
We'd like to present these alternative options for discussion and review with the DA.

Do you have a specific date in mind for the next meeting?

**Comment by godfreykutumela** [comment]
@JustusOrtlepp I suggest we table it for next week so that you can have a full 30min for this presentation. This week we have 2 items already - Reliable notification and ML Invariants.

**Comment by JustusOrtlepp** [comment]
@godfreykutumela Next week will be great, thanks. I'm assuming also Wednesday? We'll send out some more information on Monday latest.

**Comment by bushjames** [comment]
Presented and discussed at DA meeting 2021-01-20 13:00 UTC.

**Comment by johanfol** [comment]
Presentation done per minutes below.
https://community.mojaloop.io/t/da-meeting-minutes-2021-01-20-13-00-utc/254

Suggested to close this issue. Once the FRM piece is done, a new ticket will have to be raised to decide if this will form part of the ML Core (does FRM fit in the switch itself, or something standalone next to it).

**Comment by MichaelJBRichards** [comment]
JF/JO: Development has finished and this issue can be closed.
